core/src/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -83,36 +84,39 @@ internal MsalAccessTokenCacheItem
         [DataMember(Name = "expires_on", IsRequired = true)]
         internal long ExpiresOnUnixTimestamp { get; set; }
 
+        [DataMember(Name = "extended_expires_on")]
+        internal long ExtendedExpiresOnUnixTimestamp { get; set; }
+
         [DataMember(Name = "user_assertion_hash")]
         public string UserAssertionHash { get; set; }
 
         [DataMember(Name = "access_token_type")]
         internal string TokenType { get; set; }
 
-        internal string Authority
-        {
-            get
-            {
-                return string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
-            }
-        }
+        internal string Authority => string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
+        internal SortedSet<string> ScopeSet => ScopeHelper.ConvertStringToLowercaseSortedSet(NormalizedScopes);
 
-        internal SortedSet<string> ScopeSet
+        internal DateTimeOffset ExpiresOn
         {
             get
             {
-                return ScopeHelper.ConvertStringToLowercaseSortedSet(NormalizedScopes);
+                DateTime dtDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
```

Comment:
We should have a unit test here. Ideally including serialization.

Reply:
I opened https://identitydivision.visualstudio.com/DevEx/_workitems/edit/569774 to add these tests separately.

Reply:
Please see the changes Petro has made to format testing.

---
In reply to: [228273110](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1331#discussion_r228273110) [](ancestors = 228273110)